test/unit/edition/validation_test.rb

```diff
@@ -161,4 +161,15 @@ class Edition::ValidationTest < ActiveSupport::TestCase
     edition.supporting_organisations = [organisation_1]
     assert edition.valid?
   end
+
+  test 'should raise an exception when attempting to modify an edition of a locked document' do
+    edition = create(:edition)
+    edition.document.locked = true
+    edition.document.save
+
+    edition.title = 'another title'
```

Comment:
Is there any need to set the title on the edition?  If would seem the edition factory sets a title of `edition-title-n` (where `n` is the index of the edition).